src/synctree_leveldb.erl

```diff
@@ -76,17 +71,28 @@ maybe_open_leveldb(Path) ->
         [{_, DB}] ->
             {ok, DB};
         _ ->
-            ok = filelib:ensure_dir(Path),
-            {ok, DB} = safe_open(?RETRIES, Path, leveldb_opts()),
-            %% TODO: Storing LevelDB refs in ETS prevents DBs from ever
-            %%       closing. If a given node is no longer part of any
-            %%       ensembles that need a given synctree we should close
-            %%       it. For now, users will need to restart a node to
-            %%       close unneeded.
-            true = ets:insert_new(?MODULE, {Path, DB}),
-            {ok, DB}
+	    open_leveldb(Path)
     end.
 
+open_leveldb(Path) ->
+    ok = filelib:ensure_dir(Path),
+    {ok, DB} = safe_open(?RETRIES, Path, leveldb_opts()),
+    %% TODO: Storing LevelDB refs in ETS prevents DBs from ever
+    %%       closing. If a given node is no longer part of any
+    %%       ensembles that need a given synctree we should close
+    %%       it. For now, users will need to restart a node to
+    %%       close unneeded.
+    case ets:insert_new(?MODULE, {Path, DB}) of
+	true ->
+	    {ok, DB};
+	%% Another process opened the DB before we could
+	false ->
```

Comment:
If safe_open returns {ok, DB}, you are almost guaranteed to have the one true instance open on that path (except for extreme timing races), and whatever is stored in the ETS is likely stale.